lib/ansible/modules/database/postgresql/postgresql_privs.py

```diff
@@ -860,11 +936,24 @@ def main():
                 else:
                     module.warn("Role '%s' does not exist, nothing to do" % roles[0].strip())
 
+        # check if target_roles is set with type: default_privs
+        if p.target_roles and not p.type == 'default_privs':
```

Comment:
I am not sure this check is necessary, maybe add a warning of `target_roles` being ignored instead of a failure?